pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -303,6 +370,19 @@ func nextNamespaceToProvision(tcNamespaces []toolchainv1alpha1.NSTemplateSetName
 	return nil, nil, false
 }
 
+func nextNamespaceToDeprovision(tcNamespaces []toolchainv1alpha1.NSTemplateSetNamespace, namespaces []corev1.Namespace) (*corev1.Namespace, bool) {
```

Comment:
Let's add a description/comment to this method. It can be useful to have a comment describing why and what we want to deprovison here.

Reply:
done 29f23da